### ISSUE TITLE
build 429 live mode very long time before to see the configuration screen (Bugzilla Bug 1764)

### ISSUE THREAD

**Issue by tpgxyz** [OpenMandrivaAssociation/test2#1764]
This issue was created automatically with bugzilla2github
# Bugzilla Bug 1764
Date: 2016-06-18 08:44:49 +0000
From: Joel Tesson &lt;<jtsn@free.fr>&gt;
To:   OpenMandriva QA &lt;<bugs@openmandriva.org>&gt;
CC:   agoiza@gmail.com, @benbullard79, @cris-b, efremmc2@gmail.com, @itchka, @jclvanier, @rugyada, @tpgxyz, yochenhsieh@gmail.com

Last updated: 2018-08-20 18:54:49 +0000

## Comment 13826
Date: 2016-06-18 08:44:49 +0000
From: Joel Tesson &lt;<jtsn@free.fr>&gt;

Created attachment 1217
journalctl from the boot

at least 3 minutes before you see the configuration screen (language, time etc)

During this waiting I can hear my pc working at its maximum

> Attached file: journalctl.txt (text/plain, 176599 bytes)
> Description:   journalctl from the boot

## Comment 13829
Date: 2016-06-18 09:20:02 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

Created attachment 1218
htop

drakx-matchbox-winsow-manager uses 100% cpu for long time

> Attached file: foto1138.png (image/png, 14032 bytes)
> Description:   htop

## Comment 13830
Date: 2016-06-18 09:20:39 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

window

## Comment 13840
Date: 2016-06-18 14:19:49 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

may be new chinese updates?

## Comment 13862
Date: 2016-06-19 18:32:52 +0000
From: @tpgxyz

Looks like it takes 2 minutes to start for drakx's finish-install to allow user to choose lang etc.

juin 18 08:17:04 localhost.localdomain nmbd[3379]:   *****

juin 18 08:19:23 localhost.localdomain finish-install[3485]: i18n_env: lang:fr country:US locale|lang:fr_FR.UTF-8 locale|country:en_US.UTF-8 LANGUAGE:fr_FR.UTF-8:fr

## Comment 13965
Date: 2016-06-23 08:37:54 +0000
From: Joel Tesson &lt;<jtsn@free.fr>&gt;

Created attachment 1226
journalctl from boot

no change for long time, but mode live works

> Attached file: 374_live_journal.txt (text/plain, 161172 bytes)
> Description:   journalctl from boot

## Comment 14011
Date: 2016-06-27 10:37:10 +0000
From: @tpgxyz

What i've found is that modprobe fails on some *.conf file trying to resolve modules.

Can you please run:
 journalctl -b | grep modprobe

## Comment 14022
Date: 2016-06-27 11:28:46 +0000
From: Joel Tesson &lt;<jtsn@free.fr>&gt;

(In reply to Tomasz Paweł Gajc from comment #6)
> What i've found is that modprobe fails on some *.conf file trying to resolve
> modules.
> 
> Can you please run:
>  journalctl -b | grep modprobe

If you do a search on attachement 1226 you have:

juin 23 08:14:53 localhost.localdomain modprobe[2959]: modprobe: ERROR: missing parameters. See -h.

idem for attachement  1217

## Comment 14040
Date: 2016-06-27 19:15:40 +0000
From: Joel Tesson &lt;<jtsn@free.fr>&gt;

Created attachment 1234
journalctl 396 in live mode

build 396 live on hardware

always long time , I join journal (with modprobe: ERROR: missing parameters. See -h)

> Attached file: 396_live_journal.txt (text/plain, 178509 bytes)
> Description:   journalctl 396 in live mode

## Comment 14200
Date: 2016-07-09 23:39:24 +0000
From: @benbullard79

This seems still active as of build # 429.

## Comment 14276
Date: 2016-07-13 14:15:25 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

It seems that plymouth is somehow the culprit.

take a look at systemd-analize blame.

plymouth processes take more than 40s

plymouth.enable=0 passed as kernel parameter solves.

## Comment 14300
Date: 2016-07-14 11:16:56 +0000
From: @tpgxyz

When Plasma finishes loading, open konsole, log in as a root an type:

systemd-analyze blame

systemd-analyze critical-chain

Output from both commands please attach here.

## Comment 14306
Date: 2016-07-14 11:53:35 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

         41.981s plymouth-quit-wait.service
         23.968s ldconfig.service
         21.878s sddm.service
         20.118s plymouth-quit.service
          6.024s rdisc.service
          4.347s firewalld.service
          1.965s systemd-hwdb-update.service
          1.850s ModemManager.service
          1.746s accounts-daemon.service
          1.692s udisks2.service
          1.139s upower.service
           961ms nmb.service
           901ms user@0.service
           900ms polkit.service
           777ms systemd-networkd.service
           762ms fedora-readonly.service
           756ms smb.service
           734ms lvm2-monitor.service
           724ms NetworkManager.service
           684ms mdmonitor.service
           640ms fedora-loadmodules.service
           570ms winbind.service
           514ms systemd-logind.service
           509ms mandriva-everytime.service
           447ms systemd-udevd.service
           361ms systemd-journald.service
           314ms systemd-resolved.service
           284ms dev-hugepages.mount
           239ms avahi-daemon.service
           236ms nscd.service
           230ms systemd-journal-flush.service
           212ms tmp.mount
           211ms dev-mqueue.mount
           197ms systemd-sysusers.service
           172ms systemd-sysctl.service
           165ms systemd-udev-trigger.service
           160ms systemd-journal-catalog-update.service
           150ms systemd-hostnamed.service
           143ms systemd-tmpfiles-setup-dev.service
           142ms systemd-modules-load.service
           142ms sys-kernel-debug.mount
           141ms kmod-static-nodes.service
           121ms rtkit-daemon.service
           110ms systemd-tmpfiles-setup.service
           105ms systemd-timesyncd.service
           100ms systemd-update-utmp.service
            89ms systemd-random-seed.service
            80ms plymouth-read-write.service
            76ms blk-availability.service
            69ms systemd-user-sessions.service
            49ms dracut-shutdown.service
            40ms alsa-restore.service
            22ms plymouth-start.service
            16ms systemd-vconsole-setup.service
            13ms systemd-update-utmp-runlevel.service
             8ms systemd-update-done.service

## Comment 14307
Date: 2016-07-14 11:56:26 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

The time after the unit is active or started is printed after the "@" character.
The time the unit takes to start is printed after the "+" character.

graphical.target @1min 12.932s
└─[0;1;31msddm.service @51.048s +21.878s[0m
  └─[0;1;31msystemd-user-sessions.service @30.845s +69ms[0m
    └─network.target @30.842s
      └─[0;1;31mNetworkManager.service @30.118s +724ms[0m
        └─[0;1;31mfirewalld.service @25.769s +4.347s[0m
          └─[0;1;31mpolkit.service @26.373s +900ms[0m
            └─basic.target @25.531s
              └─sockets.target @25.525s
                └─nscd.socket @25.525s
                  └─sysinit.target @25.506s
                    └─[0;1;31msystemd-update-done.service @25.497s +8ms[0m
                      └─[0;1;31mldconfig.service @1.527s +23.968s[0m
                        └─local-fs.target @1.527s
                          └─local-fs-pre.target @1.527s
                            └─[0;1;31mlvm2-monitor.service @792ms +734ms[0m
                              └─dm-event.service @791ms
                                └─dm-event.socket @761ms
                                  └─-.slice

both executed starting in live mode, switching to VT, when language choose is asked

## Comment 14340
Date: 2016-07-15 23:53:21 +0000
From: @jclvanier

In live mode the logs shows that there are delays at certain point.
12 seconds here:
  juil. 13 12:51:19 localhost.localdomain kernel: r8169 0000:03:00.0 enp3s0: renamed from eth0
  juil. 13 12:51:31 localhost.localdomain rtkit-daemon[3933]: Successfully called chroot.

Then 5 + 14 + 4 seconds here:
  juil. 13 12:51:34 localhost.localdomain nmbd[4310]:   STATUS=daemon 'nmbd' : No local IPv4 non-loopback interfaces available, waiting for interface ...NOTE: NetBIOS name resolution is not supported for Internet Protocol Version 6 (IPv6).
  juil. 13 12:51:39 localhost.localdomain NetworkManager[4058]: <info>  [1468407099.3203] manager: startup complete
  juil. 13 12:51:39 localhost.localdomain systemd[1]: Failed to start rdisc daemon which discovers routers on the local subnet.
  juil. 13 12:51:53 localhost.localdomain systemd[1]: Failed to start Terminate Plymouth Boot Screen.
  juil. 13 12:51:57 localhost.localdomain acpid[4385]: starting up with netlink and the input layer
Shortly after, finish-install starts.
In this case, 33 seconds are spent to wait mainly in relation with the network.

## Comment 14341
Date: 2016-07-15 23:57:16 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

@jcl can you try with plymouth disabled?

plymouth.enable=0 passed as kernel parameter

## Comment 14347
Date: 2016-07-16 08:52:31 +0000
From: @jclvanier

Quick test that gives unexpected results. I'm puzzled.
-- without plymouth.enable=0, journalctl shows that finish-install is reached after about 51 seconds and the window asking for language pops up after 54 seconds
-- with plymouth.enable=0, journalctl shows that finish-install is reached after about 23 seconds and the window asking for language pops up after 65 seconds

## Comment 14348
Date: 2016-07-16 09:05:01 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

(In reply to jcl vanier from comment #16)
> Quick test that gives unexpected results. I'm puzzled.

With latest isos I'm puzzled too.
with plumouth.enable=0 VT console are presents, but window asking for language takes loooong time to pop up.

## Comment 14349
Date: 2016-07-16 09:07:52 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

plymouth probably is just involved in something bigger than himself :)

## Comment 14353
Date: 2016-07-16 12:21:30 +0000
From: @tpgxyz

(In reply to jcl vanier from comment #16)

> -- with plymouth.enable=0, journalctl shows that finish-install is reached
> after about 23 seconds and the window asking for language pops up after 65
> seconds

Ok this sound like finish-install is doing something nasty.

Can you boot iso into VT by adding to boot paramaeters
systemd.unit=multi-user.target

then log in as root, remove /etc/X11/xsetup.d/??finish-install.xsetup

then startx and in graphical type finish-install then attach output here

## Comment 14358
Date: 2016-07-16 15:19:19 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

(In reply to Tomasz Paweł Gajc from comment #19)
> (In reply to jcl vanier from comment #16)
> 
> > -- with plymouth.enable=0, journalctl shows that finish-install is reached
> > after about 23 seconds and the window asking for language pops up after 65
> > seconds
> 
> Ok this sound like finish-install is doing something nasty.
> 
> Can you boot iso into VT by adding to boot paramaeters
> systemd.unit=multi-user.target
> 
> then log in as root, remove /etc/X11/xsetup.d/??finish-install.xsetup
> 
> then startx and in graphical type finish-install then attach output here

in this way finish-install starts immediatly.

so I think is something that start before the culprit.

I've tried to start openbox(startx /usr/bin/openbox) and it take several seconds.
If you think it could help I can attach Xorg.log where I've notice some:

[   164.978] (II) systemd-logind: got pause for 13:66
[   164.978] (II) systemd-logind: got pause for 13:64
[   164.978] (II) systemd-logind: got pause for 13:69
[   164.978] (II) systemd-logind: got pause for 13:67
[   164.978] (II) systemd-logind: got pause for 13:68
[   164.978] (II) systemd-logind: got pause for 13:70
[   164.978] (II) systemd-logind: got pause for 13:65
[   166.128] (II) systemd-logind: got resume for 226:0

## Comment 14359
Date: 2016-07-16 15:26:47 +0000
From: @tpgxyz

(In reply to luca from comment #20)

> in this way finish-install starts immediatly.

Ok.

> so I think is something that start before the culprit.

Can you run the same procedure but instead of running finish-install execute that finish-install.xsetup script, but first add set +x after shebang

## Comment 14360
Date: 2016-07-16 15:42:37 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

(In reply to Tomasz Paweł Gajc from comment #21)
> (In reply to luca from comment #20)
> 
> > in this way finish-install starts immediatly.
> 
> Ok.
> 
> > so I think is something that start before the culprit.
> 
> Can you run the same procedure but instead of running finish-install execute
> that finish-install.xsetup script, but first add set +x after shebang

It starts immediatly in text mode.

But, as we have long boot time also in install mode, where finish-install doesn't run, may be startx or drakx-matchbox to have problems.

## Comment 14361
Date: 2016-07-16 16:12:21 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

I've done this test:

- boot in multi-user mode
- login as root
- startx /usr/bin/openbox

it takes more than 50s

- logout
- startx /usr/bin/openbox

it takes 2s

## Comment 14363
Date: 2016-07-16 18:19:02 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

I think log boot time is due to font cache build.

if I run, before startx /usr/bin/openbox, fc-cache -f -v(it needs aprox 50s) then openbox open in 2seconds.

long boot time appeared after cjk and Noto updates.

## Comment 14366
Date: 2016-07-16 20:49:30 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

removing noto-sans-cjk-font solves, at least it seems ;)
could be corrupted?

## Comment 14373
Date: 2016-07-16 22:38:56 +0000
From: @jclvanier

(In reply to luca from comment #24)
I wonder if the font cache build is done during every boot after the installation or only once during the first reboot.
In such case, that could explain why the first time I rebooted after the installation, the time to get the login window was not specially short though now I get it rather quickly.

## Comment 14374
Date: 2016-07-16 22:57:39 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

(In reply to jcl vanier from comment #26)
> (In reply to luca from comment #24)
> I wonder if the font cache build is done during every boot after the
> installation or only once during the first reboot.
> In such case, that could explain why the first time I rebooted after the
> installation, the time to get the login window was not specially short
> though now I get it rather quickly.

I think it run every time, but skipping if existing cache is valid.

## Comment 14375
Date: 2016-07-16 23:38:59 +0000
From: @tpgxyz

During iso build a font cache is being created

https://github.com/OpenMandrivaAssociation/omdv-build-iso/blob/master/omdv-build-iso.sh#L1649

## Comment 14379
Date: 2016-07-17 00:02:22 +0000
From: @tpgxyz

I added this, to build fc-cache for live user
https://github.com/OpenMandrivaAssociation/omdv-build-iso/commit/69b0a02b50ac341daacccc9c0cd932820257152a

Will run new ISO build.

## Comment 14384
Date: 2016-07-17 08:11:23 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

I've also notice that in iso build we exec ldconfig

# Rebuild linker cache
$SUDO chroot "$CHROOTNAME" /sbin/ldconfig -X

but during boot it runs again for many seconds

          23.968s ldconfig.service

## Comment 14387
Date: 2016-07-17 11:48:09 +0000
From: @jclvanier

(In reply to luca from comment #27)
Looks you are right. The timestamps of the fontcache files (/var/cache/fontconfig) is set to the installation date.
In addition, the boot time is reduced by 14s in my computer, compared with the first boot.

## Comment 14389
Date: 2016-07-17 11:55:06 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

(In reply to jcl vanier from comment #31)
> (In reply to luca from comment #27)
> Looks you are right. The timestamps of the fontcache files
> (/var/cache/fontconfig) is set to the installation date.
> In addition, the boot time is reduced by 14s in my computer, compared with
> the first boot.

I think you have to look .fonconfig present in your $HOME
is this that's missing in iso and that TPG is trying to add.

## Comment 14390
Date: 2016-07-17 11:59:21 +0000
From: @tpgxyz

(In reply to luca from comment #30)

>           23.968s ldconfig.service

This is a normals service, and it always starts, and rebuilds ldconfig cache.

Please check this ISO for long boot realted to font cache:
https://abf.openmandriva.org/platforms/3.0/products/41/product_build_lists/450

## Comment 14391
Date: 2016-07-17 12:07:08 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

(In reply to Tomasz Paweł Gajc from comment #33)
> 
> Please check this ISO for long boot realted to font cache:
> https://abf.openmandriva.org/platforms/3.0/products/41/product_build_lists/
> 450

/home/live/.fontconfig is not present

## Comment 14395
Date: 2016-07-17 12:49:22 +0000
From: @jclvanier

(In reply to luca from comment #32)
$HOME/.fontconfig is present ans contains:

-rw-rw-r-- 1 jcl 1006  584 mai    2 20:23 3830d5c3ddfd5cd38a049b759396e72e-le64.cache-6
-rw-rw-r-- 1 jcl jcl   112 juil. 16 12:12 9dae02aae2b65655eed49872589ff9b4-le64.cache-7
-rw-rw-r-- 1 jcl jcl   112 juil. 16 12:12 b4f48f7a111ce7d7fed5fea8a27d061e-le64.cache-7
-rw-r--r-- 1 jcl 1006  200 avril 26 11:01 CACHEDIR.TAG
-rw-rw-r-- 1 jcl 1006 2336 avril 26 11:01 e974607b74d59d4c8b1d173543b57c30-le64.cache-6
-rw-rw-r-- 1 jcl jcl  2344 juil. 13 13:18 e974607b74d59d4c8b1d173543b57c30-le64.cache-7

I think the files which dates are April or May  (group 1006) come from the iso.
The other are realted to gimp or libreoffice and were build after the first reboot.

/var/cache/fontconfig contains:

-rw-rw-r-- 1 root root    200 juil. 13 13:15 0251a5afa6ac727a1e32b7d4d4aa7cf0-le64.cache-7
-rw-rw-r-- 1 root root    104 juil. 13 13:14 17090aa38d5c6f09fb8c5c354938f1d7-le64.cache-7
-rw-rw-r-- 1 root root  43824 juil. 13 13:15 20b58f14c9b581391d79ea335a81488a-le64.cache-7
-rw-rw-r-- 1 root root    112 juil. 13 13:15 2d31a572ce6667f6a0da9c8dc611898b-le64.cache-7
-rw-rw-r-- 1 root root    104 juil. 13 13:14 30786aca7a961ef9f9799e540455831d-le64.cache-7
-rw-rw-r-- 1 root root  68648 juil. 13 13:15 377073e5636f5c568094b11c7a65a0a5-le64.cache-7
-rw-rw-r-- 1 root root    592 juil. 13 13:14 3830d5c3ddfd5cd38a049b759396e72e-le64.cache-7
-rw-rw-r-- 1 root root  19592 juil. 13 13:15 5d999c1bbe32f61af008974facb58b71-le64.cache-7
-rw-rw-r-- 1 root root   5248 juil. 13 13:15 69fb4bbe1b2fa860e172a32ebf28505c-le64.cache-7
-rw-rw-r-- 1 root root  43648 juil. 13 13:15 79aeb4e90a401e55ec91db207072ba77-le64.cache-7
-rw-rw-r-- 1 root root    160 juil. 13 13:15 7ef2298fde41cc6eeb7af42e48b7d293-le64.cache-7
-rw-rw-r-- 1 root root  20448 juil. 13 13:15 87f5e051180a7a75f16eb6fe7dbd3749-le64.cache-7
-rw-rw-r-- 1 root root 102928 juil. 13 13:15 8d4af663993b81a124ee82e610bb31f9-le64.cache-7
-rw-rw-r-- 1 root root    192 juil. 13 13:15 94322f4d3cdf1bd54794b691285ca062-le64.cache-7
-rw-rw-r-- 1 root root   6744 juil. 13 13:15 99a1ce9f8b6a0434aadb01d3779b0780-le64.cache-7
-rw-rw-r-- 1 root root  49200 juil. 13 13:15 a8d72120365d3c99b9746ee16d82622a-le64.cache-7
-rw-rw-r-- 1 root root  43584 juil. 13 13:15 b79f3aaa7d385a141ab53ec885cc22a8-le64.cache-7
-rw-rw-r-- 1 root root   9296 juil. 13 13:15 c2d8b2eff8efe1b2f2cfe7c403e09295-le64.cache-7
-rw-rw-r-- 1 root root  32752 juil. 13 13:15 c40c35ed428d9656ac742627e51049e6-le64.cache-7
-rw-rw-r-- 1 root root  16696 juil. 13 13:15 c776f34310d03ef601d6f862ef1a19b0-le64.cache-7
-rw-r--r-- 1 root root    200 juil.  9 00:49 CACHEDIR.TAG
-rw-rw-r-- 1 root root  40152 juil. 13 13:15 d4b6e1db2c46a3b281c413657cd2bc49-le64.cache-7
-rw-rw-r-- 1 root root  11264 juil. 13 13:15 d62e99ef547d1d24cdb1bd22ec1a2976-le64.cache-7
-rw-rw-r-- 1 root root    104 juil. 13 13:14 df311e82a1a24c41a75c2c930223552e-le64.cache-7
-rw-rw-r-- 1 root root  13224 juil. 13 13:15 e6845615947634e89d319e77806483ba-le64.cache-7
-rw-rw-r-- 1 root root    104 juil. 13 13:14 f349e9996a5320f6dd491cedd2b1f964-le64.cache-7
-rw-rw-r-- 1 root root  98600 juil. 13 13:15 f6b893a7224233d96cb72fd88691c0b4-le64.cache-7
-rw-rw-r-- 1 root root   6120 juil. 13 13:15 f9d379b867d7c69c85310a4f24e5228f-le64.cache-7

All files were build during the first reboot.
I think the reduced time to get the login window is due to this directory.

## Comment 14397
Date: 2016-07-17 12:59:05 +0000
From: @jclvanier

But yes, in the iso I tested, /home/live/.fontconfig is not present either
So what should it contain?
And why there are file dated from May or April, in my $HOME/.fontconfig?

## Comment 14398
Date: 2016-07-17 13:01:50 +0000
From: @tpgxyz

(In reply to luca from comment #34)

> /home/live/.fontconfig is not present

Try this one ISO when it finishes building
https://abf.openmandriva.org/platforms/3.0/products/41/product_build_lists/454

## Comment 14400
Date: 2016-07-17 15:03:25 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

(In reply to Tomasz Paweł Gajc from comment #37)
> (In reply to luca from comment #34)
> 
> > /home/live/.fontconfig is not present
> 
> Try this one ISO when it finishes building
> https://abf.openmandriva.org/platforms/3.0/products/41/product_build_lists/
> 454

still missing 

/root/.fontconfig is missing too so I think your latest commit doesn't work

font cache is present in /var/cache/fontconfig, but also copying it in /home/live/.fontconfig do not prevent cache rebuild

## Comment 14401
Date: 2016-07-17 15:03:54 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

Created attachment 1267
454 live home

> Attached file: foto1145.png (image/png, 42363 bytes)
> Description:   454 live home

## Comment 14402
Date: 2016-07-17 15:17:08 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

(In reply to Tomasz Paweł Gajc from comment #33)
> (In reply to luca from comment #30)
> 
> >           23.968s ldconfig.service
> 
> This is a normals service, and it always starts, and rebuilds ldconfig cache.
> 

Yes, service always start, but ldconfig runs only in case of some conditions and usually never runs.

# systemctl status ldconfig.service 
● ldconfig.service - Rebuild Dynamic Linker Cache
   Loaded: loaded (/lib/systemd/system/ldconfig.service; static; vendor preset: disabled)
   Active: inactive (dead)
Condition: start condition failed at ven 2016-07-01 00:27:02 CEST; 2 weeks 2 days ago
           none of the trigger conditions were met
     Docs: man:ldconfig(8)

# more /lib/systemd/system/ldconfig.service
#  This file is part of systemd.
#
#  systemd is free software; you can redistribute it and/or modify it
#  under the terms of the GNU Lesser General Public License as published by
#  the Free Software Foundation; either version 2.1 of the License, or
#  (at your option) any later version.

[Unit]
Description=Rebuild Dynamic Linker Cache
Documentation=man:ldconfig(8)
DefaultDependencies=no
Conflicts=shutdown.target
After=local-fs.target
Before=sysinit.target shutdown.target systemd-update-done.service
ConditionNeedsUpdate=|/etc
ConditionFileNotEmpty=|!/etc/ld.so.cache

[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=/sbin/ldconfig -X

## Comment 14403
Date: 2016-07-17 15:39:00 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

I was able to build a fonts cache in $HOME in this way:

# chroot --userspec=live:live / su live -c "/usr/bin/fc-cache -r"

## Comment 14404
Date: 2016-07-17 16:08:28 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

but, despite /var/cache/fonconfig is present and updated in the iso, I can not prevent fonts cache rebuild :( 

# fc-cache the first time rebuild the cache

## Comment 14412
Date: 2016-07-17 20:02:49 +0000
From: @tpgxyz

Please try this iso
https://abf.openmandriva.org/platforms/3.0/products/38/product_build_lists/456

I've copied /var/cache/fontrconfig to /root and home/live

## Comment 14419
Date: 2016-07-17 21:15:27 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

I've done some tests, but despite /root/.fontconfig /var/cache/fontconfig and /home/live/.fontconfig are present, the first time a cache is always rebuilded :(

## Comment 14426
Date: 2016-07-17 21:59:10 +0000
From: Joel Tesson &lt;<jtsn@free.fr>&gt;

build 456 installed

the time before calamares appears is alway very long

## Comment 14434
Date: 2016-07-17 23:08:19 +0000
From: @tpgxyz

(In reply to luca from comment #44)
> I've done some tests, but despite /root/.fontconfig /var/cache/fontconfig
> and /home/live/.fontconfig are present, the first time a cache is always
> rebuilded :(

WTF?

## Comment 14436
Date: 2016-07-17 23:09:08 +0000
From: @tpgxyz

*** Bug #1815 has been marked as a duplicate of this bug. ***

## Comment 14439
Date: 2016-07-18 06:23:25 +0000
From: Joel Tesson &lt;<jtsn@free.fr>&gt;

in bug #1815 @luca asked 

"@joel try to remove noto-sans-cjk-fonts before"

so before graphic first reboot I removed noto-sans-cjk-fonts

then rebooted in lrvel 5

what I found is that there is no more delay before the apparition of plasma

it's a normal login delay
 
to be clear I'm talking about the delay after the calamares configuration, which occurs on the first reboot.

so, to remove the doubt, TPG, can you build a test iso without noto fonts ?

and maybe the long delay before calamares appears will be also solved.

thanks

## Comment 14442
Date: 2016-07-18 09:06:25 +0000
From: @tpgxyz

(In reply to Joel Tesson from comment #48)

> so, to remove the doubt, TPG, can you build a test iso without noto fonts ?
> 

We added support for Asian languages, so removing noto-sans-cjk reverts whole work. Let's focus on finding real solution.

## Comment 14444
Date: 2016-07-18 10:13:22 +0000
From: @tpgxyz

According to this, all fontconfig options and cache has been moved to $XDG_CACHE_HOME

https://cgit.freedesktop.org/fontconfig/commit/?id=8c255fb185d5651b57380b0a9443001e8051b29d

This may explain why ~/.fontconfig is not recognized.

## Comment 14445
Date: 2016-07-18 10:47:10 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

(In reply to Tomasz Paweł Gajc from comment #50)
> According to this, all fontconfig options and cache has been moved to
> $XDG_CACHE_HOME
> 
> https://cgit.freedesktop.org/fontconfig/commit/
> ?id=8c255fb185d5651b57380b0a9443001e8051b29d
> 
> This may explain why ~/.fontconfig is not recognized.

fc-cache at least seems recognize it(look at the end):
=====================================================================
$ fc-cache -v
/usr/share/fonts: skipping, existing cache is valid: 0 fonts, 13 dirs
/usr/share/fonts/100dpi: skipping, existing cache is valid: 0 fonts, 0 dirs
/usr/share/fonts/75dpi: skipping, existing cache is valid: 0 fonts, 0 dirs
/usr/share/fonts/OTF: skipping, existing cache is valid: 0 fonts, 0 dirs
/usr/share/fonts/Speedo: skipping, existing cache is valid: 0 fonts, 0 dirs
/usr/share/fonts/TTF: skipping, existing cache is valid: 56 fonts, 7 dirs
/usr/share/fonts/TTF/antpolt: skipping, existing cache is valid: 40 fonts, 0 dirs
/usr/share/fonts/TTF/chinese: skipping, existing cache is valid: 8 fonts, 0 dirs
/usr/share/fonts/TTF/dejavu: skipping, existing cache is valid: 21 fonts, 0 dirs
/usr/share/fonts/TTF/droid: skipping, existing cache is valid: 8 fonts, 0 dirs
/usr/share/fonts/TTF/japanese: skipping, existing cache is valid: 2 fonts, 0 dirs
/usr/share/fonts/TTF/korean: skipping, existing cache is valid: 12 fonts, 0 dirs
/usr/share/fonts/TTF/liberation: skipping, existing cache is valid: 12 fonts, 0 dirs
/usr/share/fonts/Type1: skipping, existing cache is valid: 9 fonts, 0 dirs
/usr/share/fonts/cyrillic: skipping, existing cache is valid: 0 fonts, 0 dirs
/usr/share/fonts/default: skipping, existing cache is valid: 0 fonts, 2 dirs
/usr/share/fonts/default/Type1: skipping, existing cache is valid: 35 fonts, 1 dirs
/usr/share/fonts/default/Type1/adobestd35: skipping, existing cache is valid: 35 fonts, 0 dirs
/usr/share/fonts/default/ghostscript: skipping, existing cache is valid: 17 fonts, 0 dirs
/usr/share/fonts/google-crosextra-caladea: skipping, existing cache is valid: 4 fonts, 0 dirs
/usr/share/fonts/google-crosextra-carlito: skipping, existing cache is valid: 4 fonts, 0 dirs
/usr/share/fonts/misc: skipping, existing cache is valid: 59 fonts, 0 dirs
/usr/share/fonts/truetype: skipping, existing cache is valid: 0 fonts, 1 dirs
/usr/share/fonts/truetype/oxygen: skipping, existing cache is valid: 3 fonts, 0 dirs
/usr/share/fonts/ttf: skipping, existing cache is valid: 0 fonts, 2 dirs
/usr/share/fonts/ttf/decoratives: skipping, existing cache is valid: 29 fonts, 0 dirs
/usr/share/fonts/ttf/western: skipping, existing cache is valid: 9 fonts, 0 dirs
/usr/lib/X11/fonts: skipping, no such directory
/usr/X11R6/lib/X11/fonts: skipping, no such directory
/opt/ttfonts: skipping, no such directory
/home/ilprof/.local/share/fonts: skipping, no such directory
/home/ilprof/.fonts: skipping, no such directory
/var/cache/fontconfig: not cleaning unwritable cache directory
/home/ilprof/.fontconfig: cleaning cache directory
/home/ilprof/.cache/fontconfig: not cleaning non-existent cache directory
fc-cache: succeeded
=====================================================================

## Comment 14447
Date: 2016-07-18 10:52:46 +0000
From: @tpgxyz

(In reply to luca from comment #51)

> fc-cache at least seems recognize it(look at the end):

It looks for $XDG_CACHE_HOME/.fontconfig at first. Anyways i'll build new ISO to meet this change.

## Comment 14450
Date: 2016-07-18 11:24:17 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

could be the timestamp of chache files?

from man fc-cache:

-f     Force re-generation of apparently up-to-date cache files, overriding the timestamp checking.

## Comment 14454
Date: 2016-07-18 16:34:11 +0000
From: Joel Tesson &lt;<jtsn@free.fr>&gt;

458 installed no change about delays

## Comment 14528
Date: 2016-07-31 05:12:44 +0000
From: You-Cheng Hsieh &lt;<yochenhsieh@gmail.com>&gt;

If the font really bugged you so much, please remove it. I think Google droid font also contains most-used CJK font glymph, though not complete. You can advise CJK users to install the noto sans cjk font manually for better experience in the release note (or errata?).

## Comment 14565
Date: 2016-08-10 01:51:01 +0000
From: @cris-b

have looked into this and it seems to be fontconfig loading all the glyphs as part of its own initialisation.

the cache is being used here so it seems like it is a standard piece of processing.

if you run init 3 then fc-cache -sv before init 5 the delay is gone, even though the cache has not changed at all.  This is because it must be loading some stuff into memory the first time we try to load fonts (the delay has merely moved really as it just happens when running fc-cache)

so I cant see anything wrong happening here, it just seems like the sheer size of the NotoFontsCJK.ttc is causing this delay.

you will see drakx-installer-matchbox process hogging CPU at startup as it loads the glyphs.  if you disable xft and pango in matchbox then the delay moves to the startup of finish-install as this will then be the first program to initialise fontconfig.

## Comment 14566
Date: 2016-08-10 08:56:14 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

Hi Cris, and why this happens only the first boot?

## Comment 14567
Date: 2016-08-10 14:11:36 +0000
From: @cris-b

ok fair enough ;) you're right it's not normal behaviour like i first thought.

here is the answer:

when we write the cache files with fc-cache we're doing it on EXT4, the chroot we use then gets made into a squashfs filesystem which runs on the live cd.

squashfs has a timestamp resolution of 1 second, ext4 is nanosecond.

when the cache is checked the timestamp in the cache is compared to the timestamp of the directory the cache file is for.  because the timestamp in the cache is nanosecond resolution, when it looks at the timestamp for the directory it is second resolution.  The two dont match, therefore it rebuilds the cache.

this is the cause of the slow down, it's just exacerbated by the fact we now have CJK fonts which take a while to scan.

the code that does this is FcCacheTimeValid in :

https://cgit.freedesktop.org/fontconfig/tree/src/fccache.c#n555

i guess we have to find a way to ensure the timestamps match, easiest way is to write the cache with second resolution i think, rather than hacking fontconfig.

## Comment 14568
Date: 2016-08-10 14:25:07 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

Cris, I've done some tests with 503 iso and things are going better.
in vbox boot time differs only for 15/20 seconds with or without new cjk splitted fonts pkg.

also avoiding ld.cache rebuilding at first boot would increase performance. (ldconfig.service)

## Comment 14569
Date: 2016-08-10 14:38:04 +0000
From: @cris-b

yes, the ldconfig cache is probably being rebuilt for the same reason as the fontconfig one.

## Comment 14570
Date: 2016-08-10 15:09:56 +0000
From: @itchka

How on earth did you manage to figure that out; Brilliant 
Top class Crisb

## Comment 14571
Date: 2016-08-10 15:17:27 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

(In reply to Colin Close from comment #61)
> How on earth did you manage to figure that out; Brilliant 
> Top class Crisb

not for removing crisb merits, but I'm telling that from 18 July - comment 53 ;)

## Comment 14572
Date: 2016-08-11 00:33:40 +0000
From: @cris-b

building 505 with potential fixes for ldconfig and fontconfig cache rebuilding

## Comment 14573
Date: 2016-08-11 00:34:13 +0000
From: @cris-b

in progress

## Comment 14574
Date: 2016-08-11 08:17:34 +0000
From: @cris-b

looks good to me!

systemd-analyze: 20s :)

## Comment 14575
Date: 2016-08-11 08:22:26 +0000
From: luca &lt;<agoiza@gmail.com>&gt;

(In reply to crisb from comment #65)
> looks good to me!
> 
> systemd-analyze: 20s :)

tested now.
great cris  ;)  impressive boot time!!!!!!!

## Comment 14576
Date: 2016-08-11 23:31:14 +0000
From: @cris-b

thanks all, marking fixed

## Comment 14577
Date: 2016-08-11 23:31:36 +0000
From: @cris-b

fixed

## Comment 14578
Date: 2016-08-12 15:31:01 +0000
From: @jclvanier

Thanks to you all, guys :)

## Comment 14579
Date: 2016-08-12 16:47:10 +0000
From: @tpgxyz

(In reply to crisb from comment #65)
> looks good to me!
> 
> systemd-analyze: 20s :)

Great ! Thanks Cris.
